### PR TITLE
HBASE-25950 add basic compaction server metric

### PR DESCRIPTION
**Pull request by nyl3532016**
the ServerLoad has RegionLoad, ReplicationLoadSource/ReplicationLoadSink metrics, which is nothing to do with Compaction server. So,  introduce CompactionServerLoad, only has compaction related metircs

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  1s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 50s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   5m 39s |  HBASE-25714 passed  |
| +1 :green_heart: |  checkstyle  |   1m 40s |  HBASE-25714 passed  |
| +1 :green_heart: |  spotbugs  |   7m  1s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 28s |  the patch passed  |
| +1 :green_heart: |  cc  |   5m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 28s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 26s |  hbase-client: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  18m  4s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  4s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   |  68m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool |
| uname | Linux f6757d3dface 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| Max. process+thread count | 96 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 11s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   2m 57s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   8m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 20s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 52s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  2s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 34s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 207m  6s |  hbase-server in the patch passed.  |
|  |   | 249m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ac50bc95792e 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/testReport/ |
| Max. process+thread count | 3279 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m  1s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 16s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   2m 17s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   8m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 11s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 34s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 57s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 37s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 248m 59s |  hbase-server in the patch failed.  |
|  |   | 289m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b00724e24410 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/testReport/ |
| Max. process+thread count | 3261 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 37s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   5m 38s |  HBASE-25714 passed  |
| +1 :green_heart: |  checkstyle  |   1m 41s |  HBASE-25714 passed  |
| +1 :green_heart: |  spotbugs  |   6m 47s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 33s |  the patch passed  |
| +1 :green_heart: |  cc  |   5m 33s |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 33s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 41s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  17m 50s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  4s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   |  67m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 0f3ec60c5746 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 57s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   2m 51s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   8m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 20s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 58s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  3s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 34s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 206m 46s |  hbase-server in the patch passed.  |
|  |   | 248m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux f7c5e6aa1cbd 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/testReport/ |
| Max. process+thread count | 3357 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 50s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   2m 37s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   9m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 25s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 42s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 42s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 37s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 229m  4s |  hbase-server in the patch passed.  |
|  |   | 273m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 42eca599d7ac 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/testReport/ |
| Max. process+thread count | 3530 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by nyl3532016** [comment]
@Apache9 mind help review this PR,sir?

File: hbase-client/src/main/java/org/apache/hadoop/hbase/CompactionServerMetricsBuilder.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  }
+
+  private final ServerName serverName;
+  private int versionNumber;
+  private String version = "0.0.0";
+  private int infoServerPort;
+  private long compactingCellCount;
+  private long compactedCellCount;
+  private long totalNumberOfRequests;
+  @Nullable
```
Comment (Apache9) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/CompactionServerMetricsBuilder.java:73]: Nullable but we initialize it with an empty array?

File: hbase-client/src/main/java/org/apache/hadoop/hbase/CompactionServerMetricsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    private final ServerName serverName;
+    private final int versionNumber;
+    private final String version;
+    private final int infoServerPort;
+    private final long compactingCellCount;
+    private final long compactedCellCount;
+    private final List<String> compactionTasks = new ArrayList<>();
+    private final long totalNumberOfRequests;
+    @Nullable
+    private final long reportTimestamp;
```
Comment (Apache9) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/CompactionServerMetricsBuilder.java:152]: How could a long be nullable?

File: hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/master/CompactionServerListTmpl.jamon [hunk 3]
```diff
@@ -72,27 +74,18 @@ Arrays.sort(serverNames);
...
 <%java>
-    int totalRegions = 0;
-    int totalRequestsPerSecond = 0;
     int inconsistentNodeNum = 0;
     String masterVersion = VersionInfo.getVersion();
-    for (ServerName serverName: serverNames) {
 
-    ServerMetrics sl = master.getServerManager().getLoad(serverName);
+    for (ServerName serverName: serverNames) {
+    CompactionServerMetrics sl = master.getCompactionOffloadManager().getLoad(serverName);
```
Comment (Apache9) [inline hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/master/CompactionServerListTmpl.jamon:81]: Indent?

**Review by Apache9 — CHANGES_REQUESTED** [review]
No big concerns just some minor but annoying nits...

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  3s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   2m 15s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   8m 22s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 11s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 48s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 26s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  |  12m  7s |  hbase-server in the patch failed.  |
|  |   |  49m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux e77120b2d3e4 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/testReport/ |
| Max. process+thread count | 671 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   5m 50s |  HBASE-25714 passed  |
| +1 :green_heart: |  checkstyle  |   1m 49s |  HBASE-25714 passed  |
| +1 :green_heart: |  spotbugs  |   7m 25s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 45s |  the patch passed  |
| +1 :green_heart: |  cc  |   5m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 45s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 27s |  hbase-client: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  19m 53s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   |  71m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 2ad3f5c549cd 4.15.0-128-generic #131-Ubuntu SMP Wed Dec 9 06:57:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25714 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 20s |  HBASE-25714 passed  |
| +1 :green_heart: |  compile  |   3m 23s |  HBASE-25714 passed  |
| +1 :green_heart: |  shadedjars  |   9m 22s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 32s |  HBASE-25714 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 11s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 11s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m  9s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 11s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 57s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 222m 39s |  hbase-server in the patch passed.  |
|  |   | 268m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3350 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 311558186e50 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25714 / 997af0215c |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/testReport/ |
| Max. process+thread count | 3032 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3350/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**